src/de.rs

```diff
@@ -78,7 +85,7 @@ impl<'de, 'a, R: Read> de::Deserializer<'de> for &'a mut Deserializer<R> {
     where
         V: Visitor<'de>,
     {
-        visitor.visit_u8(self.rdr.read_u8()?)
+        visitor.visit_u8(self.rdr.bin_read_integer()?)
```

Comment:
Shouldn't it be bin_read_bool: https://github.com/ChainSafe/serde-bin-prot/blob/2a5e469870dfdeb91dc7e2d0e2590fa77dc2eee6/src/read_ext.rs#L21

Reply:
Nope that's line 50. I think it is a quirk of how the diff viewer is displaying it.